common/getpass.go

```diff
@@ -0,0 +1,46 @@
+package common
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+)
+
+// GetPass attempts to read a file for a password at the supplied location.
+// If it fails, then the original supplied string will be returned to the caller.
+// A valid string will always be returned, regardless of whether an error occurred.
+func GetPass(s string, idx int) (string, error) {
```

Comment:
What do you think of just moving this into `common/util.go` or `eth/helpers` instead of its own file?

Reply:
If you think that's best, we can do that...  I put it into its own file because I knew I would be writing tests and I prefer to keep things in separate files when each function may have multiple permutations to test.

Reply:
Actuallly I thought about this some more and what we should do is the following

In livepeer.go , just pass `*ethPassphrase` to `client.Setup` https://github.com/livepeer/go-livepeer/blob/dc878b5f938b0c7c7fdf5e69bccf3122c78e78b4/cmd/livepeer/livepeer.go#L352

So no changes have to actually be made here (except perhaps altering the `-ethPassword` flag description). 

In `eth/accountmanager.go` there is an `Unlock()` function, this is where we also prompt for the passphrase in case the `-ethPassword` flag was not provided on node startup.  This is the method that's being called in `client.Setup()` to unlock the account. 

My suggestion is that we therefore move this helper into `eth/helpers` so that we can use it in `eth/accountManager.Unlock()`. 

In addition we can change the current default behaviour of the password prompt to also be able to take in a path (this is where we will change from how geth does is). This means that the user can always supply a password or path regardless of whether the `-ethPassword` flag is used. 

`AccountManager` really is our specific layer on top of what go-ethereum provides so I think it makes sense to implement it there instead of `cmd` or `common`. WDYT?

Reply:
I'm not sure I agree with this -- this same function will be used for orchSecret.

Reply:
I'm not sure I agree with this -- this same function will be used for orchSecret.  With that in mind, does it still make sense to make the changes you're proposing?

Reply:
> I'm not sure I agree with this -- this same function will be used for orchSecret.

Good point, did not think ahead here. We can leave it in `common` then but I'd have a slight preference having it in `util.go` then, but I'll leave that decision up to you.

As for the rest of the changes described I think it's worth to implement them, we can import the helper from the `common` package then. 

We can do something like 

```
func(am *accountManager) Unlock(passphrase string) error {
	if passphrase == "" {
		glog.Infof("Please enter the passphrase or path to password file to unlock Ethereum account %v", am.account.Address.Hex())
		var err error
		passphrase, err = getPassphrase(false)
		if err != nil {
			return err
		}
	}

	pass, err := common.GetPass(passphrase)
	if err != nil {
		return err
	}
	err = am.keyStore.Unlock(am.account, pass)
	if err != nil {
		return err
	}

	am.unlocked = true

	glog.Infof("Unlocked ETH account: %v", am.account.Address.Hex())

	return nil
}
```

Reply:
Done.  Please check :-)

As far as moving `GetPass` into `common/util` , it doesn't really go along with the spirit of [Idiomatic Golang](https://about.sourcegraph.com/go/idiomatic-go) ... On the other hand, neither do some of the other parts of this repository.

I hope we'll have a chance to clean up and refactor this sort of thing before too much time goes by.  As an open source project, we should strive to be more idiomatic.

Meanwhile, it's also important to stick with the style of existing code.  ...

It's clear how I feel about this, but I'll move it if that's your preference.

Reply:
I don't think it's necessary for the interactive prompt for passphrase to be able to read from a file.  The point of this was to keep the cleartext passphrase out of the command line for security purposes.

Reply:
> I don't think it's necessary for the interactive prompt for passphrase to be able to read from a file.

You're probably right.

Changes look good, can we clean up the commit history ?

Reply:
Best practice is to squash on merge.  There is no need to flatten the commits in the PR itself.